lib/snowflex/db_connection/server.ex

```diff
@@ -0,0 +1,156 @@
+defmodule Snowflex.DBConnection.Server do
```

Comment:
I wonder if some of the code in this could be shared with `Snowflex.Worker` (maybe move the common code into a shared module that they both use?). I'm not super familiar with the application, though, so it may make more sense to keep them separate, just a thought.